GSoC: Updating the map

MarinStevic opened the pull request:
This is an update to the current nodewatcher map. I have added:
- Fulscreen option
- Popup with some general information about the node that you get when you click on it. And also by clicking the name in the popup you can go to that nodes page
- Sidebar that gives you a list of all currently online nodes with a search bar and the ability to locate each node on the map
- Sidebar also shows all nodes that have gone offline in the past 24 hours
- Updated the markercluster.js to work with the new leaflet.js
- Leaflet markercluster and fullscreen are now gathered through CDN so they will be the newest version

CdavM commented on nodewatcher/modules/frontend/map/static/leaflet/css/leaflet.fullscreen.css:
```diff
@@ -0,0 +1,40 @@
+    }
+
+/* Do not combine these two rules; IE will break. */
+.leaflet-container:-webkit-full-screen {
```
Where is this used?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/code.js:
```diff
@@ -3,6 +3,20 @@
+		
+		var panelContent = {
+			id: 'nodeList',
+			tab: '<i class="fa fa-bars" style="margin-top: 6px;"></i>',
```
Why are we hard-coding HTML styles? And hard coding html tags too?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/code.js:
```diff
@@ -3,6 +3,20 @@
         var detail = e.originalEvent ? e.originalEvent.detail : e.detail;
         var map = detail.map;
 
+		map.addControl(new L.Control.Fullscreen());
```
Why is this block indented?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/code.js:
```diff
@@ -3,6 +3,20 @@
+			id: 'nodeList',
+			tab: '<i class="fa fa-bars" style="margin-top: 6px;"></i>',
+			title: 'Node list',
+			//position: 'bottom'
```
What is this comment for?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+function sidebarTableCreate(){
+    var leafletSidebarDiv = document.querySelector('div.leaflet-sidebar-content');
+	var sidebarSearch = document.createElement('input');
```
Is this indented more than the previous line? if so, why?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+	sidebarSearch.id = "myInput";
+	sidebarSearch.placeholder = "Search for nodes...";
+	sidebarSearch.title = "Type in a name";
+	sidebarSearch.style.width = "100%";
```
Will this styling also look good on mobile?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+}
+
+function sidebarTableAddNode(node){
+	var sidebarTable  = document.getElementById('node-list-table');
```
What if this ID changes? Why are we hard-coding this?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+	
+	var sidebarTableCell_name = sidebarTableRow.insertCell();
+	var a = document.createElement('a');
+	a.href = '../node/' + node.data.i;
```
What if the link changes? Can this be made more robust?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+	sidebarTableRow.className = "node-list-table-row";
+	
+	var sidebarTableCell_name = sidebarTableRow.insertCell();
+	var a = document.createElement('a');
```
Can we use a more informative variable name than "a"?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+	sidebarTableCell_name.appendChild(a);
+	
+	var sidebarTableCell_type = sidebarTableRow.insertCell();
+	var type = document.createElement("p");
```
I'm not sure what "type" refers to. Can we make this var name more descriptive?

CdavM commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,106 @@
+		sidebarTable.appendChild(sidebarTableRow);
+	}
+	else {
+		for (i = 1; i < (rows.length); i++) {
```
I think indentation is inconsistent across this PR, please fix (with whatever is the current standard in nodewatcher)

CdavM commented on nodewatcher/modules/frontend/map/templates/map/map.html:
```diff
@@ -9,6 +9,16 @@
     {% add_data "js_data_end" "leaflet/js/leaflet.markercluster.js" %}
     {% add_data "css_data" "leaflet/css/leaflet.markercluster.css" %}
 
+	{% add_data "js_data_end" "leaflet/js/leaflet.fullscreen.js" %}
```
Indentation?

CdavM commented on nodewatcher/modules/frontend/map/templates/map/map.html:
```diff
@@ -19,7 +29,7 @@
 {% block content %}
     {% leaflet_map "map" %}
-
+	
```
?

CdavM commented on nodewatcher/modules/administration/location/static/location/js/map.js:
```diff
@@ -5,8 +5,10 @@
-            return L.marker({lng: node.data.l[0], lat: node.data.l[1]});
+        if (node.data.l) {
+            sidebarTableAddNode(node,"node-list-table");  //Add the node to the sidebar active node list
+            return L.marker({lng: node.data.l[0], lat: node.data.l[1]}).bindPopup('Name: <a href="../node/' + node.data.i + '">' + node.data.n + '</a><br>Type: ' + node.data.t);  //Show a popup when a node is selected with it's name and type
```
typo: it's -> its

mitar commented on nodewatcher/modules/frontend/map/templates/map/map.html:
```diff
@@ -6,23 +6,99 @@
-    {% add_data "css_data" "leaflet/css/leaflet.markercluster.css" %}
+    {% addtoblock "js" %}<script src='https://unpkg.com/leaflet.markercluster@1.3.0/dist/leaflet.markercluster-src.js'></script>{% endaddtoblock %}
+    {% addtoblock "css" %}<link href='https://unpkg.com/leaflet.markercluster@1.3.0/dist/MarkerCluster.Default.css' rel='stylesheet' />{% endaddtoblock %}
+	{% addtoblock "css" %}<link href='https://unpkg.com/leaflet.markercluster@1.3.0/dist/MarkerCluster.css' rel='stylesheet' />{% endaddtoblock %}
```
Keep the same indent.

mitar replied:
Also, I would prefer if we add files directly to the repo instead of fetching them from CDNs. I know it might be slower, but then it works also without Internet access. Which for community networks does happen. (Map is not really the best example, because it will not work without Internet, but it is a pattern I hope we could keep.)

mitar commented on nodewatcher/modules/frontend/map/static/map/js/sidebar.js:
```diff
@@ -0,0 +1,80 @@
+//Add nodes to the map sidebar table
```
Code style in this file is strange. Some functions have four spaces indentation, some two. Please make it the same as other JS files in the project.

MarinStevic committed:
GSoC: Updating the map

This is an update to the current nodewatcher map. I have added:

-Fulscreen option
-Popup with some general information about the node that you get when you click on it. And also by clicking the name in the popup you can go to that nodes page
-Sidebar that gives you a list of all currently online nodes with a search bar and the ability to locate each node on the map
-Sidebar also shows all nodes that have gone offline in the past 24 hours
-Updated the markercluster.js to work with the new leaflet.js
-Leaflet markercluster and fullscreen are now gathered through CDN so they will be the newest version
